src/modules/endpoint/sagas/watchFetchLanguagesForCategory.js

```diff
@@ -0,0 +1,51 @@
+// @flow
+
+import type { Saga } from 'redux-saga'
+import { call, put, takeLatest } from 'redux-saga/effects'
+import type {
+  FetchCategoryActionType,
+  FetchLanguagesForCategoryActionType,
+  FetchLanguagesForCategoryFailedActionType,
+  PushLanguagesActionType
+} from '../../app/StoreActionType'
+import type { DataContainer } from '../DataContainer'
+import loadLanguages from './loadLanguages'
+
+function * fetchLanguages (dataContainer: DataContainer, action: FetchLanguagesForCategoryActionType): Saga<void> {
+  const {city, language, forceUpdate} = action.params
+  try {
+    yield call(dataContainer.setContext, city, language)
+    yield call(loadLanguages, city, dataContainer, forceUpdate)
+
+    const languages = yield call(dataContainer.getLanguages)
+
+    const insert: PushLanguagesActionType = {
+      type: `PUSH_LANGUAGES`,
+      params: {
+        languages,
+        city,
+        language
+      }
+    }
+    yield put(insert)
+
+    if (languages.map(language => language.code).includes(language)) {
+      const fetchCategory: FetchCategoryActionType = {
```

Comment:
better put this action in separate file like ` src/modules/endpoint/actions/index.js` and import them here

Reply:
I guess we don't do that for the whole project @a-kud… It's also not too bad, since we use our FlowTypes to create actions, so they're at least well-shaped.
We could discuss if we should add action-creators (which I would rather decline), but that's sth for a separate task...